src/rust/engine/options/src/parse.rs

```diff
@@ -80,41 +86,44 @@ peg::parser! {
             = quiet!{ "]" }
             / expected!("the end of a list indicated by ']'")
 
-        rule list_items() -> Vec<String>
+        rule list_items<T>(parse_value: rule<T>) -> Vec<T>
             = list_start()
-            items:quoted_string() ** ","
-            whitespace()* ","? whitespace()*
+            items:value(&parse_value) ** ","
+            ","? whitespace()*
             list_end() {
                 items
             }
 
-        rule items() -> Vec<String>
+        rule items<T>(parse_value: rule<T>) -> Vec<T>
             = whitespace()*
-            items:(tuple_items() / list_items())
+            items:(tuple_items(&parse_value) / list_items(&parse_value))
             whitespace()* { items }
 
-        rule list_edit() -> ListEdit<String>
-            = whitespace()* action:action() items:items() whitespace()* {
+        rule list_edit<T>(parse_value: rule<T>) -> ListEdit<T>
+            = whitespace()* action:action() items:items(&parse_value) whitespace()* {
                 ListEdit { action, items }
             }
 
-        rule list_edits() -> Vec<ListEdit<String>>
-            = e:list_edit() ** "," ","? { e }
+        rule list_edits<T>(parse_value: rule<T>) -> Vec<ListEdit<T>>
+            = e:list_edit(&parse_value) ** "," ","? { e }
 
-        rule list_replace() -> Vec<ListEdit<String>>
-            = items:items() {
+        rule list_replace<T>(parse_value: rule<T>) -> Vec<ListEdit<T>>
+            = items:items(&parse_value) {
                 vec![ListEdit { action: ListEditAction::Replace, items }]
             }
 
-        rule implicit_add() -> Vec<ListEdit<String>>
+        rule implicit_add<T>(parse_raw_value: rule<T>) -> Vec<ListEdit<T>>
             // If the value is not prefixed with any of the syntax that we recognize as indicating
             // our list edit syntax, then it is implicitly an Add.
-            = !(whitespace() / (action() list_start()) / (action() tuple_start()) / tuple_start() / list_start()) item:string() {
-                vec![ListEdit { action: ListEditAction::Add, items: vec![item.to_owned()] }]
+            = !(whitespace() / (action() list_start()) / (action() tuple_start()) / tuple_start() / list_start()) item:parse_raw_value() {
+                vec![ListEdit { action: ListEditAction::Add, items: vec![item] }]
             }
 
+        pub(crate) rule int_list_edits() -> Vec<ListEdit<i64>>
+            = implicit_add(<integer()>) / list_replace(<integer()>) / list_edits(<integer()>)
```

Comment:
Is it worth packaging up the sequence of `implicit_add / list_replace / list_edits` calls into their own helper too? (I guess it might need two parser arguments to support the `unquoted_string` vs. `quoted_string` difference)

Reply:
It might be for "everything that isn't a string", since only string has that quoting issue. But that can wait for one of the several followups.